intuita-webview/src/jobDiffView/DiffViewer/index.tsx

```diff
@@ -25,17 +25,18 @@ export const JobDiffView = ({
 			? 'inline'
 			: 'side-by-side';
 	});
-	const [isVisible, setVisible] = useState(false);
+	const [isVisible, setVisible] = useState(true);
 
-	const toggleViewed = useCallback(() => setVisible((v) => !v), [setVisible]);
-
-	useEffect(() => {
-		if (isVisible) {
-			collapsableRef.current?.expand();
-		} else {
-			collapsableRef.current?.collapse();
-		}
-	}, [isVisible]);
+	const toggleViewed = useCallback(() => {
```

Comment:
what was wrong here? also, you have no guarantee that the `setVisible` function's callback is going to be run once, it's should not have any side effects

Reply:
```jsx
	useEffect(() => {
		if (isVisible) {
			collapsableRef.current?.collapse();
		} else {
			collapsableRef.current?.expand();
		}	
	}, [isVisible])
```

runs on the initial mount, so even though we initialise `isVisible` to `true`, the views were collapsed. If we want to stick with `useEffect`, we can do

```jsx
	const mountedRef = useRef(false);

	useEffect(() => {
		if (!mountedRef.current) {
			return;
		}
		if (isVisible) {
			collapsableRef.current?.expand();
		} else {
			collapsableRef.current?.collapse();
		}	
	}, [isVisible])

	useEffect(() => {
		mountedRef.current = true;
	}, [])
```